### PR TITLE
Added new Multimodal AI Block utilizing openrouter's API Service.

### PR DESCRIPTION
**Pull request by ymrohit**
This PR introduces the MultimodalAIBlock class, a new block that integrates multimodal AI models via OpenRouter API. It allows users to input a text prompt and an image URL, and select from several multimodal models for analysis. The block processes the inputs and returns a response based on the selected AI model.

### CODE REVIEW

**Comment by aarushik93** [comment]
thanks for the PR @ymrohit - just curious there's no speech modality models available via Openrouter as of now right?

**Comment by ymrohit** [comment]
@aarushik93 I don't think there is any speech modality models available via openrouter at the moment.

**Review by aarushik93 — APPROVED** [review]

File: autogpt_platform/infra/helm/autogpt-market/values.prod.yaml [hunk 1]
```diff
@@ -93,7 +93,7 @@ env:
...
+  DATABASE_URL: "AgCe47Y6QFnZRmNNpc3iVIcJ9zAZMPr/1OU61Slr0oM1yjneT7LgNTrAD9+QSu7Q0q2u+533FImOu8o/pKG0gNzEtyeSEEsegmCiaxs4Wi+/jVodU8CgfrYN3xkgi7QuD2fMs/zz+sv25LDg8LR1mIvkv+W/PkbqoVx3UAUXL7ZM+UvkFqie+69C/kwoH51xPaYi+FajSzql+VdhDc28aFkwrDApenyLbyKAHSnplWqpFDVtcQK4dL7K9EhRkUnWORNHiYT+N0Ras5tw1ief481nDud29nhlQ+5HIHGK5QouKmA1yJIJqavbzsDf7n+nWCwULIWXGi5d8A23e8vZAuF2CInUHAmP8d+xBDRx8aATZigoOM0GpHn5ljxNs0TCZ1HvQ8TjzQOpcgmBZ9l5Eod3y/PRhjGBRrcR550PAxpkjRj9VjfMN5NSiHVsGw7xKQr2+4yghv3VgSZFvOg1qUxmleJ9khwZq2Gel0uvhGStYw50x/C+EDMaJNQUaBDDbo5Bjy6/RNAEgrX0snQAZWCZSg4bsb4+2jRvo50W1wMr9076qOFViNn2YM+exIGXdmgggAKk7foZDZz1KgG+dAhKBXoUhXwHO9P450gq4xxDGSk0ZCe10OiszNkXVtmmE+Ryi5JjnbDsxLJP1tynrljFUb29VePt0zwZAhdnz0MQU1T9IVbjcKYB92h19lhcdgE="
```
Comment (aarushik93) [inline autogpt_platform/infra/helm/autogpt-market/values.prod.yaml]: please remove

File: autogpt_platform/backend/backend/util/settings.py [hunk 2]
```diff
@@ -250,7 +250,7 @@ class Secrets(UpdateTrackingModel["Secrets"], BaseSettings):
     replicate_api_key: str = Field(default="", description="Replicate API Key")
     unreal_speech_api_key: str = Field(default="", description="Unreal Speech API Key")
     ideogram_api_key: str = Field(default="", description="Ideogram API Key")
-
+    openrouter_api_key: str = Field(default="", description="OpenRouter API Key")
```
Comment (aarushik93) [inline autogpt_platform/backend/backend/util/settings.py:253]: Please migrate to new credentials system

**Review by aarushik93 — CHANGES_REQUESTED** [review]
Please see comments

**Comment by aarushik93** [comment]
@ymrohit we've now added OpenRouter to the new platform and so this PR is no longer needed....we'd still love for you to contribute other blocks and integrations 🙏🏽

**Pull request closed without merging**